File: e-project-4_api/src/main/java/com/example/e_project_4_api/dto/response/UserResponse.java
package com.example.e_project_4_api.dto.response;

import com.example.e_project_4_api.models.Artists;
import com.example.e_project_4_api.models.FavouriteSongs;
import com.example.e_project_4_api.models.Playlists;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Collection;
import java.util.Date;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

public class UserResponse {

    private Integer id;
    private String username;
    private String fullName;
    private String avatar;
    private String password;
    private String phone;
    private String email;
    private String role;
    private String bio;
    private Date dob;
    private Boolean isDeleted;
    private Boolean isActive;
    private Date createdAt;
    private Date modifiedAt;
    private Integer artistId;

}

File: dto/user_response.dart
class UserResponse {
  final int id;
  final String username;
  final String fullName;
  final String avatar;
  final String password;
  final String phone;
  final String email;
  final DateTime? dob;

  UserResponse({
    required this.id,
    required this.username,
    required this.fullName,
    required this.avatar,
    required this.password,
    required this.phone,
    required this.email,
    this.dob,
  });

  factory UserResponse.fromJson(Map<String, dynamic> json) {
    return UserResponse(
      id: json['id'],
      username: json['username'],
      fullName: json['fullName'],
      avatar: json['avatar'],
      password: json['password'],
      phone: json['phone'],
      email: json['email'],
      dob: DateTime.parse(json['dob']),
    );
  }
}
